spec/features/global_announcements_spec.rb

```diff
@@ -0,0 +1,57 @@
+require 'rails_helper'
+
+RSpec.describe 'Global announcements', type: :feature do
+  subject { page }
+  let(:instance) { create(:instance) }
+
+  with_tenant(:instance) do
+    describe 'no global announcements' do
+      before do
+        instance.announcements.clear
```

Comment:
> I've seen projects grow and I've seen code bases evolve. There's always unquantifiable changes that happen to code bases over time and I'm trying to make sure as few ways these things can happen.
@lowjoel I think next time we should sit down together to further discuss this and I think I'll learn more from you, I'm still not fully aware about the things you said here :(   So for now you don't need to adopt my solution.

@weiqingtoh Thanks for the detailed comments ! >< Actually I think abstraction might not have impact on code readability cause it's the Rspec's conventional way and it improves maintainability. Well, same, next time we can discuss further offline, let's end the discussion here.